backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -342,9 +343,17 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
         saveBranch(
             session, commitEntity.getCommit_hash(), ModelDBConstants.MASTER_BRANCH, repository);
       }
-      session.getTransaction().commit();
       if (create) {
-        createRoleBindingsForRepository(request, userInfo, repository);
+        try {
+          createRoleBindingsForRepository(request, userInfo, repository);
+          repository.setDeleted(false);
+          session.update(repository);
+        } catch (Exception ex) {
+          session.delete(repository);
+          throw ex;
+        } finally {
+          session.getTransaction().commit();
+        }
```

Comment:
we should rely on the deletion cron to clean up.

Reply:
Removed from here and leave it for cron job